samples/nrf_desktop/src/modules/config.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdlib.h>
+
+#include <zephyr.h>
+#include <settings/settings.h>
+
+#include "event_manager.h"
+#include "config_event.h"
+
+#define MODULE config
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_LOG_LEVEL);
+
+#define KEY_LEN 9
+
+/* Array of pointers to data loaded from settings */
+static u8_t *loaded_data[CONFIG_EVENT_ID_MAX];
+
+static bool config_id_is_supported(u8_t id)
+{
+	if (IS_ENABLED(CONFIG_DESKTOP_MOTION_OPTICAL_ENABLE)) {
```

Comment:
If we keep it in section we would not have to add a conditional compilation check when something new is added.
Let's keep it for now as we need to have it in today. I will check how to fix it next week.

Reply:
this conditional compilation groups supported ids in code, it's positive thing IMHO